test/e2e/run.sh

```diff
@@ -765,6 +765,17 @@ verify() { # script API
     #          nodes and that pod0c0 is allowed to run on 4 CPUs:
     #   verify 'set.intersection(nodes["pod0c0"], nodes["pod1c0"]) == set()' \
     #          'len(cpus["pod0c0"]) == 4'
+    local retries=5
```

Comment:
#1063 wraps implement an explicit configurable `retry` and wraps some of the `verify` calls in that. Does not make sense to do both, so we need to chose either. I am fine with both approaches. If a retrying verify is often necessary then perhaps this approach is slightly preferred. But maybe we should make it parametrizable with something like this:
```bash
verify() {
    local retries=${retries:-1} poll_delay=${poll_delay:=0s}
    # if we want to default to a retrying verify then we just need to set the defaults up so, eg.
    # local retries=${retries:-5} poll_delay=${poll_delay:=0.2s}
    while (( retries > 0 )); do
        ( __verify "$@" ) && return 0
        retries=$(( retries - 1 ))
        echo "re-trying verify $@, $retries tries left..."
        sleep $poll_delay
    done
    error "failed to verify $@"
}

# Then you can do a non-retrying/default verify like
verify 'len(cpus["pod0c0"]) == 4'
# and a corresponding retrying verify with something like
retries=5 poll_delay=0.2s verify 'len(cpus["pod0c0"]) == 4'
```

Reply:
> If a retrying verify is often necessary then perhaps this approach is slightly preferred

This is always needed. Not just in one test

Reply:
@marquiz No, I don't think it is always necessary. For instance for most basic workload placement tests it should not be needed. There you typically schedule a few pods (waiting for them to be running) then test allocations. I think in those cases by the time the shell regains control after the CRI requests returns everything should be done on the runtime side. The asynchronous race should be present only in cases when you react to a policy decision making event/action, for instance based on a logged message in the coldstart case, and then start checking resources...

Anyway, regardless of this I'm fine with always retrying by default.

Reply:
OK, maybe so. Anyway, I don't want to spend time (and I'm not sure anyone else does) on combing all the test assertions to figure out whether `--retry` is possibly needed or not. Just have retry everywhere by default.  It shouldn't do any harm or cause extra delays. Then add `--retry 0` to those cases you're absulutely sure that retries won't be necessary (and shouldn't happen)